src/Microsoft.PowerShell.Commands.Management/commands/management/Service.cs

```diff
@@ -689,6 +689,207 @@ protected override void ProcessRecord()
         }
 
         #endregion Overrides
+
+        /// <summary>
+        /// Adds UserName, Description, BinPath and Delay Autostart info to the ServiceController object.
+        /// </summary>
+        /// <param name="service"></param>
+        /// <returns></returns>
+        private PSObject AddProperties(ServiceController service) {
+            NakedWin32Handle hScManager = IntPtr.Zero;
+            NakedWin32Handle hService = IntPtr.Zero;
+            int lastError = 0;
+            PSObject serviceAsPSObj = PSObject.AsPSObject(service);
+            try
+            {
+                hScManager = NativeMethods.OpenSCManagerW(
+                    lpMachineName: service.MachineName,
+                    lpDatabaseName: null,
+                    dwDesiredAccess: NativeMethods.SC_MANAGER_CONNECT
+                );
+                if (IntPtr.Zero == hScManager) {
+                    lastError = Marshal.GetLastWin32Error();
```

Comment:
Seems like by the time this is called, we know `service` is valid so I suppose you can reasonably assume any failures are due to Access Denied (however, I didn't look at all the possible return values from the win32 apis).  However, I think you should at least add a comment about why you're not checking the `lastError` code.  Applies to the other ones below as well.

Reply:
I'm not checking `lastError` because the documentation says `If the function fails, the return value is NULL.` [Source][0] and also because the other service classes called the function the same way.

I thought it would be better to have the potential exception speak for itself rather than assuming that the error is a Access Denied.

But I might be wrong. If I am, please tell me.

[0]: https://msdn.microsoft.com/en-us/library/windows/desktop/ms684323(v=vs.85).aspx

Reply:
Thanks for the explanation, I'm ok with this.